File: LeetCode/Medium/Jump Game II/C/jump_game_two.c
#define MAX(a,b) (((a) > (b)) ? (a) : (b))

int jump(int* nums, int numsSize){
    int curr_far = 0, curr_end = 0;
    int res = 0;
    for (int i=0; i<numsSize-1; i++) {
        curr_far = MAX(curr_far, i+*(nums+i));
        if (i == curr_end) {
            res++;
            curr_end = curr_far;
        }
    }
    return res;
}

File: LeetCode/Medium/Jump Game II/Java/jumpGameTwo.java
class Solution {
    public int jump(int[] nums) {
        int res = 0;
        int currFar = 0, currEnd = 0;
        int n = nums.length;
        for (int i=0; i<n-1; i++) {
            currFar = Math.max(currFar, i+nums[i]);
            if (i == currEnd) {
                System.out.println(i + " " + currFar);
                res++;
                currEnd = currFar;
            }
        }
        return res;
    }
}
